pkg/instance/service.go

```diff
@@ -97,6 +97,38 @@ func (s service) SaveInstance(instance *model.DeploymentInstance) error {
 	return s.instanceRepository.SaveInstance(instance)
 }
 
+func (s service) DeleteInstance(deploymentId, instanceId uint) error {
+	deployment, err := s.FindDeploymentById(deploymentId)
+	if err != nil {
+		return err
+	}
+
+	index := slices.IndexFunc(deployment.Instances, func(instance *model.DeploymentInstance) bool {
+		return instanceId == instance.ID
+	})
+	if index == -1 {
+		return errdef.NewNotFound("instance not found on deployment(%d) by id: %d", deployment.ID, instanceId)
+	}
+
```

Comment:
The next line belongs to the above `slices.IndexFunc` and find the instance logic. I would remove the newline to indicate that.